nautobot/ipam/tables.py

```diff
@@ -247,26 +247,35 @@ class VRFDeviceAssignmentTable(BaseTable):
         linkify=lambda record: record.vrf.namespace.get_absolute_url(),
         accessor="vrf.namespace.name",
     )
-    device = tables.Column(
-        linkify=lambda record: record.device.get_absolute_url(), accessor="device.name", verbose_name="Device"
+    related_object_type = tables.TemplateColumn(
+        template_code="""
+        {% if record.device %}
+            Device
+        {% elif record.virtual_machine %}
+            Virtual Machine
+        {% else %}
+            Virtual Device Context
+        {% endif %}
+        """
     )
-    virtual_machine = tables.Column(
-        linkify=lambda record: record.virtual_machine.get_absolute_url(),
-        accessor="virtual_machine.name",
-        verbose_name="Virtual Machine",
-    )
-    virtual_device_context = tables.Column(
-        linkify=lambda record: record.virtual_device_context.get_absolute_url(),
-        accessor="virtual_device_context.name",
-        verbose_name="Virtual Device Context",
+    related_object_name = tables.TemplateColumn(
+        template_code="""
+        {% if record.device %}
+            <a href="{{ record.device.get_absolute_url }}">{{ record.device.name }}</a>
+        {% elif record.virtual_machine %}
+            <a href="{{ record.virtual_machine.get_absolute_url }}">{{ record.virtual_machine.name }}</a>
+        {% else %}
+            <a href="{{ record.virtual_device_context.get_absolute_url }}">{{ record.virtual_device_context.name }}</a>
+        {% endif %}
```

Comment:
Would something like this work?

```
{% load helpers %}
{{ record.device or record.virtual_machine or record.virtual_device_context | hyperlinked_object }}
```

Reply:
It did not work here. I also tried using `with` tag, it didn't work either
```
{% load helpers %}
        {% with related_object=record.device or record.virtual_machine or record.virtual_device_context %}
            {{ related_object | hyperlinked_object }}
        {% endwith %}
```